### ISSUE TITLE
Implement Sparse KronProd

### ISSUE THREAD

**Issue by Jbwasse2** [tauhid03/MarkovChainAnalysis#10]
Currently the implementation of sparse kronprod seems to work on 99% of cases. Find a small version of the failure case and fix the bug.

**Comment by Jbwasse2** [comment]
The previous implementation was done in python2, so I need to update this to python3.

**Comment by Jbwasse2** [comment]
The problem with the mismatch in hitting time results was actually from a bug in the regular hitting time code. When forming the KronProd object, we originally took the markov matrices as 
```
    Plist = (Mlist)
    A = KronProd(Plist)
```

when it should have been

```
    Plist = list(reversed(Mlist))
    A = KronProd(Plist)
```

**Comment by Jbwasse2** [comment]
The sparse code is fast, on the example we are using it manages to finish in 1215 steps while the non sparse code is at iteration 475. (They should both have the same number of iterations).

**Comment by Jbwasse2** [comment]
Implemented in #12
